client/signup/steps/woocommerce-install/transfer.tsx

```diff
@@ -10,6 +22,76 @@ interface Props {
 export default function Transfer( { goToStep }: Props ): ReactElement | null {
 	const { __ } = useI18n();
 
+	const siteId = useSelector( getSelectedSiteId ) as number;
+
+	const [ progress, setProgress ] = useState( 0 );
+	const [ error, setError ] = useState( { transferFailed: false, transferStatus: null } );
+
+	const dispatch = useDispatch();
+
+	// Select state
+	const fetchingTransferStatus = !! useSelector( ( state: AppState ) =>
+		isFetchingAutomatedTransferStatus( state, siteId )
+	);
+	const transferStatus = useSelector( ( state: AppState ) =>
+		getAutomatedTransferStatus( state, siteId )
+	);
+
+	const transferFailed = useSelector( ( state: AppState ) => hasUploadFailed( state, siteId ) );
+
+	// Initiate Atomic transfer
+	useEffect( () => {
+		if ( ! siteId ) {
+			return;
+		}
+
+		dispatch( fetchAutomatedTransferStatus( siteId ) ); // todo: can possibly drop this, polling should be kicked off by initiate call
+		dispatch( initiateThemeTransfer( siteId, null, 'woocommerce' ) );
+	}, [ siteId, dispatch ] );
+
+	// Watch transfer status
+	useEffect( () => {
+		if ( fetchingTransferStatus ) {
+			return;
+		}
+
+		switch ( transferStatus ) {
+			case transferStates.NONE:
+			case transferStates.PENDING:
+			case transferStates.INQUIRING:
+			case transferStates.PROVISIONED:
+			case transferStates.FAILURE:
```

Comment:
are `FAILURE`, `REVERTED`, `CONFLICTS` states that represent 20% of the whole process?

Reply:
oh, I see... it's handled below. 😶

Reply:
> represent 20% of the whole process?

No this is something that needs improving, I just made it up for illustrative purposes. 

The actual request is extremely slow. Some discussion about the request speed here. p1634705094195400-slack-C7YPW6K40